src/tools/illink/src/ILLink.Tasks/build/Microsoft.NET.ILLink.targets

```diff
@@ -38,6 +38,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <EnableUnsafeBinaryFormatterSerialization Condition="'$(EnableUnsafeBinaryFormatterSerialization)' == ''">false</EnableUnsafeBinaryFormatterSerialization>
     <EnableUnsafeUTF7Encoding Condition="'$(EnableUnsafeUTF7Encoding)' == ''">false</EnableUnsafeUTF7Encoding>
     <BuiltInComInteropSupport Condition="'$(BuiltInComInteropSupport)' == ''">false</BuiltInComInteropSupport>
+    <ValueTypeTrimFriendlySelect Condition="'$(ValueTypeTrimFriendlySelect)' == ''">true</ValueTypeTrimFriendlySelect>
```

Comment:
Do you have any measurements how this affects `PublishTrimmed`? This enables it if `PublishTrimmed` is true, I wonder if it should set defaults based on `PublishAot` instead.